e2e/config/config.yaml

```diff
@@ -33,25 +33,33 @@ cron_job:
 store:
   path: ./e2e/testdb
 
-oracle:
-  etherscan:
-    enable: true
-    api_key: 473P55YZMBNSRR63YF78D2PMMIUYUQJMGY
-    apis:
-      gas_price: https://api.etherscan.io/api?module=gastracker&action=gasoracle&apikey=
-  polygonscan:
-    enable: true
-    api_key: BZM7P395BQS1YFKQMA2AK2ACQKWGEW4JB3
-    apis:
-      gas_price: https://api.polygonscan.com/api?module=gastracker&action=gasoracle&apikey=
-  coinmarketcap:
+conversion_rate_oracles:
+  - implementation: coinmarketcap
+    source: coinmarketcap
     enable: true
+    url: https://pro-api.coinmarketcap.com/v2/cryptocurrency/quotes/latest?
     api_key: 1408daf0-0777-4916-9fe4-20da5ee77560
-    apis:
-      query_rate: https://pro-api.coinmarketcap.com/v2/cryptocurrency/quotes/latest?
 
-gas_price_domains:
-  [ ethereum, polygon ]
+# domainIDs with gas price oracle mapping
+gas_price_oracles_with_domain_ids:
```

Comment:
this is where I got confused and struggled, like I mentioned yesterday, fee oracle is a oracle oriented services, so that it should have a list of supported oracles as the top logic, but now domain is on top of oracle logically, which affects another type of oracle -- conversion rate oracle, so I have to identify them separately, however, conversion rate oracle has nothing to do with domain, if I rename here to domains, how can I name converstion rate oracle?

Reply:
Conversion rate APIs are not domain specific. Gas price APIs are domain specific.
Conversion rate APIs can be left as a separate list on its own.

Reply:
and you are correct, in my understanding, fee oracle should have supported oracles as the first class citizen, followed by the type of oracles, in our case we have two now, 1) gas price oracle, 2)conversion rate oracle, followed by that each type of oracle have their interface. this is the structure I have, now you suggested that put domain on top of one type of oracle, which for sure is going to achieve what you want, but it will break this logic structure. lets say if a new hiring saw the fee oracle config for the first time, he would be very confused why there is a domain list, I hope what I said makes sense

Reply:
I understand what you mean I just don't agree with it.
Also, please let's remove all mentions of oracle from APIs it is really idiomatically wrong.
As a service it is an oracle, but everything else are just APIs so they are not really first class citizens.

Reply:
Design wise here we are working with information about tokens and domains.
Everything else it dependent on those two.
Where we fetch these information is second priority.